cub/cub/device/dispatch/tuning/tuning_select_if.cuh

```diff
@@ -511,6 +518,232 @@ struct sm90_tuning<__uint128_t, flagged::yes, keep_rejects::yes, offset_size::_4
+  static constexpr int threads                       = 256;
+  static constexpr int items                         = 20;
+  static constexpr BlockLoadAlgorithm load_algorithm = BLOCK_LOAD_WARP_TRANSPOSE;
+  static constexpr CacheLoadModifier load_modifier   = LOAD_DEFAULT;
+  using delay_constructor                            = detail::exponential_backon_constructor_t<516, 685>;
+};
+
+template <class Input>
+struct sm100_tuning<Input, flagged::no, keep_rejects::no, offset_size::_4, primitive::yes, input_size::_2, may_alias::yes>
+{
+  // trp_1.ld_0.ipt_20.tpb_384.ns_1060.dcid_5.l2w_375 1.109871  0.973142  1.105415  1.459135
+  static constexpr int threads                       = 384;
+  static constexpr int items                         = 20;
+  static constexpr BlockLoadAlgorithm load_algorithm = BLOCK_LOAD_WARP_TRANSPOSE;
+  static constexpr CacheLoadModifier load_modifier   = LOAD_DEFAULT;
+  using delay_constructor = detail::exponential_backon_jitter_window_constructor_t<1060, 375>;
+};
+
+template <class Input>
+struct sm100_tuning<Input, flagged::no, keep_rejects::no, offset_size::_4, primitive::yes, input_size::_4, may_alias::no>
+{
+  // trp_1.ld_0.ipt_15.tpb_384.ns_1508.dcid_5.l2w_585 1.201993  0.920103  1.185134  1.441805
+  static constexpr int threads                       = 384;
+  static constexpr int items                         = 15;
+  static constexpr BlockLoadAlgorithm load_algorithm = BLOCK_LOAD_WARP_TRANSPOSE;
+  static constexpr CacheLoadModifier load_modifier   = LOAD_DEFAULT;
+  using delay_constructor = detail::exponential_backon_jitter_window_constructor_t<1508, 585>;
+};
+
+template <class Input>
+struct sm100_tuning<Input, flagged::no, keep_rejects::no, offset_size::_4, primitive::yes, input_size::_4, may_alias::yes>
+{
+  // trp_1.ld_0.ipt_19.tpb_512.ns_928.dcid_7.l2w_770 1.258815  1.000000  1.235251  1.444884
+  static constexpr int threads                       = 512;
+  static constexpr int items                         = 19;
+  static constexpr BlockLoadAlgorithm load_algorithm = BLOCK_LOAD_WARP_TRANSPOSE;
+  static constexpr CacheLoadModifier load_modifier   = LOAD_DEFAULT;
+  using delay_constructor                            = detail::exponential_backon_constructor_t<928, 770>;
+};
+
+template <class Input>
+struct sm100_tuning<Input, flagged::no, keep_rejects::no, offset_size::_4, primitive::yes, input_size::_8, may_alias::no>
+    // baseline remained the fastest
+    : sm90_tuning<Input, flagged::no, keep_rejects::no, offset_size::_4, primitive::yes, input_size::_8>
+{};
+
+template <class Input>
+struct sm100_tuning<Input, flagged::no, keep_rejects::no, offset_size::_4, primitive::yes, input_size::_8, may_alias::yes>
+{
+  // trp_1.ld_0.ipt_23.tpb_384.ns_1140.dcid_7.l2w_520 1.081506  0.955298  1.088848  1.248971
+  static constexpr int threads                       = 384;
+  static constexpr int items                         = 23;
```

Comment:
The tuning permutes the following parameters:

```c++
constexpr int NOMINAL_4B_ITEMS_PER_THREAD = TUNE_ITEMS_PER_THREAD; // <<< changed by tuning
constexpr int ITEMS_PER_THREAD = // << ends up being used by the agent
      CUB_MIN(NOMINAL_4B_ITEMS_PER_THREAD, CUB_MAX(1, (NOMINAL_4B_ITEMS_PER_THREAD * 4 / sizeof(InputT))));
```

While encoding the selected tuning (`trp_1.ld_0.ipt_23.tpb_384.ns_1140.dcid_7.l2w_520`) what should've ended up in the encoding was `min(23, max(1, 23 * 4 // 8))` which is `11`.

Reply:
Fixed